### PR TITLE
Upstreaming internal development

### PR DESCRIPTION
**Pull request by ts-amz**
# Description
Upstreaming internal development improvements.
Updated format of service, client, and integ files.

Changes include:
- most endpoint 500s changed to their proper 400 level errors
- reduce the information on error responses (full data is available in logs)
- set automatic pagination limits on all endpoints to remove DOS vector through unpaginated calls
- update aws-sdk dependencies to align on version 2.1386.0
- add additional validation on numerous endpoints
- add additional checks to integration tests to validate the environment is ready for testing

## Type of change

- [x] *Bug fix (non-breaking change which fixes an issue)*
- [x] *New feature (non-breaking change which adds functionality)*
- [ ] *Breaking change (fix or feature that would cause existing functionality to not work as expected)*
- [ ] *Refactor* (existing code being refactored)
- [ ] *This change includes a documentation update*

# Submission Checklist

- [ ] CI dry-run passing
- [ ] Integration tests passing locally
- [ ] Change logs generated

### CODE REVIEW

**Review by canavandl — APPROVED** [review]
I spent ~3 hrs looking at the changes and this looks as reasonable as anything else. The lint changes are pretty easy to scan and the actual upstreamed changes match my understanding.

**Comment by aaronatbissell** [comment]
This looks like a lot of formatting/non-functional changes. Is there anything functionally changing as a part of this PR? It's tough to look through 1200+ files.

**Comment by ts-amz** [comment]
> This looks like a lot of formatting/non-functional changes. Is there anything functionally changing as a part of this PR? It's tough to look through 1200+ files.

The PR is broken into commits that represent chunks of work. However, this required some bundling of commits. I've updated the description with the high level view of the changes, but going through the pre-formatting commits is the best way to see what has changed.

Within the formatting commits, there are only minor changes to simplify or improve integration testing.

Thanks!

File: source/packages/integration-tests/src/step_definitions/notifications/events.steps.ts [hunk 1]
```diff
@@ -57,6 +56,7 @@ const eventsourcesService: EventsourcesService = container.get(
 );
 
 Given('I am using event {string}', async function (name: string) {
+    // logger.debug(`I am using event '${name}'`);
```
[the remark this answers is not in the document]
Comment (ts-amz) [reply source/packages/integration-tests/src/step_definitions/notifications/events.steps.ts:59]: These kinds of comments existed before and were commented out by a team member in a particular commit. I left them to not further complicated patching/merging. Happy to uncomment them or remove them in a later commit (I'm partial to keep them and setting the appropriate log level).

**Review by BenjiTheC — APPROVED** [review]

**Pull request merged**